### DISCUSSION TITLE
certificate has unsupported critical extension

### DISCUSSION THREAD

**Discussion by 1218100112** [Keyfactor/ejbca-ce#127] [Q&A]
May I ask why this error was reported, Certificate has unsupported Critical Extension, when an external CA was used to create a CA

**Comment by primetomas** [comment]
Then you first has to tell us when you encounter this message. If it's even in EJBCA?

**Reply by primetomas** [reply]
Technically, the reason is clear. A Certificate contained a Critical extension that the parsing software does not understand. That is the definition of Criticality in certificates (See rfc5280), if an extension is critical it needs to be understood, if an extension is non-critical, the parsing software can safely ignore it if not understood.

**Reply by 1218100112** [reply]
thanks，but we have configured the corresponding extensions in the Custom Extensions option. If Custom extensions can be used in the profile of the issued certificate, why can't custom extensions be used in the External CA? The system does not support it. Or do we have a configuration problem?

**Reply by 1218100112** [reply]
Or a version problem

**Reply by primetomas** [reply]
This is not an EJBCA configuration, how you configure extensions in EJBCA does not affect standard RFC5280 validation processing. You should basically never make a custom extension that is _critical_. 

Can you tell us more about what custom extension you have added? 

I recommend simply removing the critical checkbox, unless you have a standard specificasion stating it must be critical.

**Reply by 1218100112** [reply]
This is the content stipulated in the CCC specification. do you mean that custom extensions cannot use critical？

**Reply by primetomas** [reply]
Can you link to this specification? I'm merely guessing due to the lack of details here.
